contracts/mainnet/metaregistry/MetaRegistry.vy

```diff
@@ -110,13 +110,26 @@ def _get_pool_from_lp_token(_token: address) -> address:
 @internal
 @view
 def _get_registry_handler_from_pool(_pool: address) -> address:
+    """
+    @dev sometimes a factory pool can be registered in a manual registry
+         because of this, we always take the last registry a pool is
+         registered in and not the first, as manual registries are first
+         and factories come later
+    """
```

Comment:
Well it can't be agnostic, can it. The previous code had the following logic

```
for i in range(MAX_REGISTRIES):
        if i == self.registry_length:
            break
        handler: address = self.get_registry[i].registry_handler
        if self.get_registry[i].is_active and RegistryHandler(handler).is_registered(_pool):
            return handler
```

so, if it finds a pool in a registry, return that registry handler. So if a pool that is a factory pool (there are a few) is registered in a registry (which in the addressprovider is always behind the factory so far) then the older logic would return the registry handler and not the factory registry handler (which is actually where the pool was registered/deployed). So, we always want the last one.

There is a guarantee that registry will be listed in the address provider before factory, so I think the solution proposed will be best for these cases (where pool is registered in registry as well as factory).

Reply:
My point wasn't about the registries but about the registry handlers. There is no logic to enforce the order in which registry handlers are added. So someone could add a Factory Registry Handler before a Stable Registry Handler and the Metaregistry should be agnostic to that. Which it already is, but this comment suggests that it isn't / shouldn't. And that's misleading because it can.

It's also not technically true that the factory pool is always behind the registry pool in the address provider. Factory pools do have a higher index in the address provider, but the entry for the corresponding manual registry could be inactive or point to address zero (if the id is unset). 

It's also possible that the factory contract that was used to deploy the pool was superseded by a new one where the old pool is not registered. There are, iirc, actually a few cases like this on mainnet where old factory pools are on the stable registry but are not on the newer factory pool contract. 

To be clear, I'm not saying this logic is better or worse than the previous one. IMO both are fine, the metaregistry just needs to point to *something* and handle it properly. I am just suggesting editing or removing the comment to not suggest that there is a preferred/valid order, because we do not enforce one.

Reply:
One thing we could do is for `_get_registry_handler_from_pool`, we could do `_get_registry_handlers_from_pool` such that it returns all registry handlers:

```
@internal
@view
def _get_registry_handlers_from_pool(_pool: address) -> address[MAX_REGISTRIES]:
    """
    @dev sometimes a factory pool can be registered in a manual registry
         because of this, we always take the last registry a pool is
         registered in and not the first, as manual registries are first
         and factories come later
    """

    pool_registry_handler: address[MAX_REGISTRIES] = empty(address[MAX_REGISTRIES])
    for i in range(MAX_REGISTRIES):

        if i == self.registry_length:
            break
        handler: address = self.get_registry[i].registry_handler

        if self.get_registry[i].is_active and RegistryHandler(handler).is_registered(_pool):
            pool_registry_handler[i] = handler

    if pool_registry_handler[0] == ZERO_ADDRESS:
        raise("no registry")
    return pool_registry_handler
```
That way, we could amend the contract to query using this `handler_id`:

```
@external
@view
def get_admin_balances(_pool: address, _handler_id: uint256 = 0) -> uint256[MAX_COINS]:
    """
    @notice Get the current admin balances (uncollected fees) for a pool
    @dev _handler_id < 1 if pool is registry in one handler, more than 0 otherwise
    @param _pool Pool address
    @param _handler_id id of registry handler
    @return List of uint256 admin balances
    """
    return RegistryHandler(self._get_registry_handlers_from_pool(_pool)[_handler_id]).get_admin_balances(_pool)
```